src/Features/LanguageServer/Protocol/RoslynLanguageServerTarget.cs

```diff
@@ -0,0 +1,123 @@
+            _listener = listenerProvider.GetListener(FeatureAttribute.LanguageServer);
+            _serverKind = serverKind.ToConvertableString();
+
+            // Create services that require base dependencies (jsonrpc) or are more complex to create to the set manually.
+            var lifeCycleManager = new RoslynLifeCycleManager(this);
+            _baseServices = GetBaseServices();
+            _serviceCollection = GetServiceCollection(jsonRpc, this, logger, capabilitiesProvider, lifeCycleManager, serverKind.ToConvertableString(), supportedLanguages);
+        }
+
+        protected override ILspServices ConstructLspServices()
+        {
+            return _lspServiceProvider.CreateServices(_serverKind, _baseServices, _serviceCollection);
+        }
+
+        private IServiceCollection GetServiceCollection(
+            JsonRpc jsonRpc,
+            IClientCapabilitiesProvider clientCapabilitiesProvider,
+            IRoslynLspLogger logger,
+            ICapabilitiesProvider capabilitiesProvider,
+            RoslynLifeCycleManager lifeCycleManager,
+            string serverKind,
+            ImmutableArray<string> supportedLanguages)
+        {
+            var serviceCollection = new ServiceCollection()
+                .AddSingleton<IClientLanguageServerManager>(new ClientLanguageServerManager(jsonRpc))
+                .AddSingleton<ILspLogger>(logger)
+                .AddSingleton<IRoslynLspLogger>(logger)
+                .AddSingleton<IClientCapabilitiesProvider>(clientCapabilitiesProvider)
+                .AddSingleton<ICapabilitiesProvider>(capabilitiesProvider)
+                .AddSingleton<LifeCycleManager<RequestContext>>(lifeCycleManager)
+                .AddSingleton(new ServerInfoProvider(serverKind, supportedLanguages))
+                .AddSingleton<IRequestContextFactory<RequestContext>, RequestContextFactory>()
+                .AddSingleton<IRequestExecutionQueue<RequestContext>>((serviceProvider) => GetRequestExecutionQueue())
+                .AddSingleton<IClientCapabilitiesManager, ClientCapabilitiesManager>();
+
+            return serviceCollection;
+        }
+
+        private static ImmutableArray<Lazy<ILspService, LspServiceMetadataView>> GetBaseServices()
+        {
+            var baseServices = ImmutableArray.Create<Lazy<ILspService, LspServiceMetadataView>>();
+
+            return baseServices;
+        }
+
+        protected override void RequestExecutionQueueErroredInternal(string message)
+        {
+            var messageParams = new LogMessageParams()
+            {
+                MessageType = MessageType.Error,
+                Message = message
+            };
+            var lspServices = (LspServices)GetLspServices();
+            var clientNotificationService = lspServices.GetRequiredLspService<IClientLanguageServerManager>();
+
+            var asyncToken = _listener.BeginAsyncOperation("RequestExecutionQueue_Errored");
+            _errorShutdownTask = Task.Run(async () =>
```

Comment:
as mentioned above this needs to be moved to the base to cover this and shutdown and exit.

Reply:
I instead moved this logic into the LifeCycleManager. If that doesn't address your concern lets chat offline so I can understand the goals of this task better.